src/wp-includes/link-template.php

```diff
@@ -4520,15 +4521,17 @@ function get_avatar_data( $id_or_email, $args = null ) {
 		'r' => $args['rating'],
 	);
 
-	if ( is_ssl() ) {
-		$url = 'https://secure.gravatar.com/avatar/' . $email_hash;
-	} else {
-		$url = sprintf( 'http://%d.gravatar.com/avatar/%s', $gravatar_server, $email_hash );
-	}
+	/*
+	 * Gravatars are always served over HTTPS.
+	 *
+	 * The Gravatar website redirects HTTP requests to HTTPS URLs so always
+	 * use the HTTPS scheme to avoid unnecessary redirects.
+	 */
+	$url = 'https://secure.gravatar.com/avatar/' . $email_hash;
```

Comment:
This seems to be fine. Have you had any thoughts on how this might this affect `'get_gravatar_url'` filters?

Reply:
> Have you had any thoughts on how this might this affect `'get_gravatar_url'` filters?

I haven't, do you have any concerns?

Previously the URLs haven't been consistent: either secure... or xx.... but it's possible there will be an effect if `is_ssl() === false`. Maybe that will need a paragraph in the developer changes dev note.

Reply:
I don't have any definitive concerns at the moment, other than the theoretical filtering of `'get_gravatar_url'` where `$args['scheme']` is checked, and possibly even used, where a site isn't using HTTPS, and it tries to do something with an expected `http://` URL that is now `https://`.

Reply:
I understand `$args['scheme']` being retained for `'pre_get_avatar_data'` filters. Should the value be set to `'https'` once the URL is changed to `https://`, or not?

Reply:
Noting that `$args` is documented from here onwards as:
> Arguments passed to `get_avatar_data()`, **after processing**.

**my emphasis**

Reply:
I considered that but eventually decided not to.

For plugins replacing the gravatar URL with a local version, setting the scheme so it doesn't match the parsed value/`is_ssl()` default could lead to problems on servers without an SSL certificate.

These days that's most likely a dev server but it would be concerning if all of a sudden the dev site didn't work.

Reply:
@costdev I asked Dion for his input via PM and he thought it was fine to leave as is (ie, pass the scheme to the filter as used in the arguments or the default value).

Let me know if you consider it a blocker or are happy to go with passing it unchanged.

Reply:
That works for me!